src/bandwidth_limit.cpp

```diff
@@ -59,18 +59,27 @@ namespace libtorrent {
 
 	void bandwidth_channel::update_quota(int dt_milliseconds)
 	{
+		TORRENT_ASSERT_VAL(m_limit >= 0, m_limit);
+
+		TORRENT_ASSERT_VAL(m_limit < inf, m_limit);
+
 		if (m_limit == 0) return;
 
-		// avoid integer overflow
-		if (m_limit >= std::numeric_limits<int>::max() / dt_milliseconds)
+		// "to_add" should never have int64 overflow: "m_limit" contains < "<int>::max"
+		std::int64_t to_add = (m_limit * dt_milliseconds + 500) / 1000;
```

Comment:
actually, it seems weird that ``m_limit`` would be a 64 bit int, if it can only hold values <= ``INT32_MAX``.. it seems like it would make more sense to widen these calculations instead..